frontend/src/App.tsx

```diff
@@ -11,15 +12,20 @@ const paths = [
   { element: <HomePage />, url: PATH_TO_HOMEPAGE },
   { element: <GamePage />, url: PATH_TO_GAMEPAGE },
 ];
-
-const App: FC = (): ReactElement => (
-  <BrowserRouter>
-    <Routes>
-      {paths.map((path) => (
-        <Route key={path.url} path={path.url} element={path.element} />
-      ))}
-    </Routes>
-  </BrowserRouter>
-);
+const App: FC = (): ReactElement => {
+  const { pathname } = useLocation();
+  const pathsWithoutHeader = [PATH_TO_GAMEPAGE];
+  return (
+    <div className="h-screen bg-DARKTHEME_BACKGROUND_COLOR ">
+      {pathsWithoutHeader.indexOf(pathname) >= 0 ? null : <Header />}
+      <Routes>
+        {paths.map((path) => (
+          <Route key={path.url} path={path.url} element={path.element} />
+        ))}
+      </Routes>
+      {pathsWithoutHeader.indexOf(pathname) >= 0 ? null : <Castle />}
```

Comment:
same scenario as with the header - you can go even further and create variable instead of calling this method 2 times